builds/fabric8-che/plugins/keycloak-plugin-token-provider/src/main/java/com/redhat/che/keycloak/token/provider/oauth/OpenShiftGitHubOAuthAuthenticator.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat inc.
+
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat - Initial Contribution
+ *******************************************************************************/
+package com.redhat.che.keycloak.token.provider.oauth;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.eclipse.che.api.auth.shared.dto.OAuthToken;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.security.oauth.GitHubOAuthAuthenticator;
+
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.util.store.MemoryDataStoreFactory;
+
+@Singleton
+public class OpenShiftGitHubOAuthAuthenticator extends GitHubOAuthAuthenticator {
+
+    @Inject
+    public OpenShiftGitHubOAuthAuthenticator(@Nullable @Named("che.oauth.github.clientid") String clientId,
+                                             @Nullable @Named("che.oauth.github.clientsecret") String clientSecret,
+                                             @Nullable @Named("che.oauth.github.redirecturis") String[] redirectUris,
+                                             @Nullable @Named("che.oauth.github.authuri") String authUri,
+                                             @Nullable @Named("che.oauth.github.tokenuri") String tokenUri,
+                                             @Named("che.oauth.github.forceactivation") boolean forceActivation) throws IOException {
+
+        super(clientId, clientSecret, redirectUris, authUri, tokenUri);
+
+        if (forceActivation &&
+            !isNullOrEmpty(authUri) &&
+            !isNullOrEmpty(tokenUri) &&
+            redirectUris != null && redirectUris.length != 0) {
+
+            configure("NULL", "NULL", redirectUris, authUri, tokenUri, new MemoryDataStoreFactory());
```

Comment:
maybe we could just remove this "forceActivation" as in rh-che it would be always activated